### PR TITLE
feat: `cleanKv()` and ULIDs

### LINKED ISSUE #148: Expose methods to allow running cronjobs against session's expiry time
I have enjoyed using this library for handling sessions. However I have noticed that at some point there will be sessions that don't get deleted from the the store and so overtime could grow and will need a way to delete them.

As I understand the Kv store API currently doesn't have and true TTL capabilities natively but I think it should be possible to still expose a way to search/filter sessions by expiry time.

The official docs state this:

```md
- A range selector selects all keys that are lexicographically between the given start and end keys (including the start, and excluding the end). For example, the selector ["users", "a"], ["users", "n"] will select all keys that start with the prefix ["users"] and have a second key part that is lexicographically between a and n, such as ["users", "alice"], ["users", "bob"], and ["users", "mike"], but not ["users", "noa"] or ["users", "zoe"].
```

Based on this I think if a key included a timestamp (could be `[PREFIX, timestamp]`) then could expose a function like `getExpiredSessions` that may look like this:

```ts
const db = await Deno.openKv();
const entries = db.list({ start: [KV_PREFIX, 0], end: [KV_PREFIX, unix_timestamp] })
for await (const entry of entries) {
  entry.key; // [KV_PREFIX, expiry_as_unix]
  entry.value; // { sessionId, expiryAsUnix, ...rest }
  entry.versionstamp; // "00000000000000010000"
}
```

I'm unsure on the `deleteExpiredSessions` method but I think it would need to use transactions for consistency.

Be interested to open the dialogue to see other opinions on the matter.

I saw this bit of [code](https://github.com/nhrones/kvdbClient/blob/master/lib/kvdb.ts#L80) and felt it may be a good starting point

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #148

### CODE REVIEW

**Comment by iuioiua** [comment]
Now that KV entries use ULIDs in their keys, the previous KV data must be deleted for the migration. The `kv` instance is deliberately closed off in this module. Our options for a migration:
1. Have `cleanKv()` have an `expiredOnly = true` argument. When `false`, all KV data is deleted (which only includes data for this module).
2. Export the `kv` instance as part of this module and provide a simple migration script in the README. Said differently, is there any benefit to having the `kv` instance kept private?

File: README.md [hunk 1]
```diff
@@ -128,6 +130,16 @@ provider you like.
    GITHUB_CLIENT_ID=xxx GITHUB_CLIENT_SECRET=xxx deno run --unstable --allow-env --allow-net server.ts
    ```
 
+1. Clean-up expired KV entries as part of a cron job, if possible.
+
+   ```ts
+   import { cleanKv } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
```
Comment (kt3k) [inline README.md:136]: `cleanKv` might sound confusing as what's cleared are not obvious from the name. How about `cleanExpired` or similar?

**Comment by iuioiua** [comment]
> Now that KV entries use ULIDs in their keys, the previous KV data must be deleted for the migration. The `kv` instance is deliberately closed off in this module. Our options for a migration:
> 
> 1. Have `cleanKv()` have an `expiredOnly = true` argument. When `false`, all KV data is deleted (which only includes data for this module).
> 2. Export the `kv` instance as part of this module and provide a simple migration script in the README. Said differently, is there any benefit to having the `kv` instance kept private?
> 3. Refactor the repo so it's BYO `kv` instance. Previously, I avoided this to minimize how much the dev has to think about.

WDYT, @kt3k and @losfair?

**Comment by kt3k** [comment]
> 1. Have cleanKv() have an expiredOnly = true argument. When false, all KV data is deleted (which only includes data for this module).

This looks fine. Alternatively how about providing API like `clearLegacySessions()` or `migrate` to do the migration?

> 2. Export the kv instance as part of this module and provide a simple migration script in the README. Said differently, is there any benefit to having the kv instance kept private?
> 3. Refactor the repo so it's BYO kv instance. Previously, I avoided this to minimize how much the dev has to think about.

I don't understand these options well. Why does passing kv instances matter?

**Comment by iuioiua** [comment]
[Key expiration is coming to Deno KV!](https://github.com/denoland/deno/pull/20091) Should we instead wait for that to land?

**Comment by kt3k** [comment]
I think using native key expiration would be much easier and cleaner. I'm not sure when it's available in Deploy though.

**Comment by iuioiua** [comment]
Ok. Let's wait for that.

**Pull request closed without merging**